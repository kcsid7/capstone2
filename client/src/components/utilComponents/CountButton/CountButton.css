.CountButton{
    display: flex;
}


.CountButton-Count {
    text-align: center;
    border: none;
    box-shadow: 3px 3px 3px black, -1px -1px black;
}

.CountButton-Button {
    width: 40px;
    font-size: 1.5rem;
    margin: 0 0.3rem;
    border-radius: 50%;
}

.CountButton-Increase:hover, .CountButton-Decrease:hover {
    transform: translate(0, -5px);
    cursor: pointer;
    
}

.CountButton-Increase {
    border: none;
    background-color: rgb(31, 210, 31);
    box-shadow: 3px 3px 3px rgb(1, 66, 1), -0.1px -1px 2px green;
    color: white;

}

.CountButton-Decrease {
    border: none;
    background-color: rgb(210, 46, 46);
    box-shadow: 3px 3px 3px rgb(66, 1, 1), -0.1px -1px 2px red;
    color: white;

}

