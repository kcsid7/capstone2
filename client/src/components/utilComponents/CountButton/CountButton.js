import React, {useState} from "react";

//MUI

import "./CountButton.css"

function CountButton({count, toggleCount}) {

    return(
        <div className="CountButton">
            <button 
                className="CountButton-Increase CountButton-Button" 
                onClick={() => toggleCount("UP")}>+</button>
            <input 
                type="text" 
                readOnly
                value={count} 
                style={{backgroundColor: `${(count > 0) ? "lightgreen" : "white"}`}}
                className="CountButton-Count CountButton-Button"/>
            <button 
                className="CountButton-Decrease CountButton-Button" 
                disabled={count <= 0}
                onClick={() => toggleCount("DOWN")}>-</button>
        </div>
    )
}

export default CountButton;