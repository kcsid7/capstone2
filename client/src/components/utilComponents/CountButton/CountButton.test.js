import React from "react";
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom'; 

import CountButton from "./CountButton";


// Smoke Test
test("Renders without crashing", () => {
    render(<BrowserRouter><CountButton /></BrowserRouter>)
})


test("Count Button snapshot", () => {
    const { asFragment } = render(<BrowserRouter><CountButton /></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
})