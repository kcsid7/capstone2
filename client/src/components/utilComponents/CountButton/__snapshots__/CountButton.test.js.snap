// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count Button snapshot 1`] = `
<DocumentFragment>
  <div
    class="CountButton"
  >
    <button
      class="CountButton-Increase CountButton-Button"
    >
      +
    </button>
    <input
      class="CountButton-Count CountButton-Button"
      readonly=""
      style="background-color: white;"
      type="text"
      value=""
    />
    <button
      class="CountButton-Decrease CountButton-Button"
    >
      -
    </button>
  </div>
</DocumentFragment>
`;
